pkg/vsphere/disk/disk_manager.go

```diff
@@ -275,24 +281,40 @@ func (m *Manager) Detach(op trace.Operation, d *VirtualDisk) error {
 		return errors.Trace(err)
 	}
 
-	spec := types.VirtualMachineConfigSpec{}
-
-	disk, err := findDisk(op, m.vm, d.DatastoreURI)
+	disk, err := findDiskByFilename(op, m.vm, d.DatastoreURI)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
+	if err = m.detach(op, disk); err != nil {
+		op.Errorf("detach for %s failed with %s", d.DevicePath, errors.ErrorStack(err))
+		return errors.Trace(err)
+	}
+
+	func() {
```

Comment:
Ya know... that really is idiotic.  Thanks for pointing it out.  I have no idea why it exists.  Will fix.